src/lib/common/JsonHelper.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+*
+* Copyright 2013 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms
+* of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Orion dev team
+*/
+#include "common/JsonHelper.h"
+
+std::string toJsonString(const std::string& input)
+{
+  std::ostringstream ss;
+  ss << '"';
+  for (std::string::const_iterator iter = input.begin(); iter != input.end(); ++iter) {
+        switch (*iter) {
+            case '\\': ss << "\\\\"; break;
+            case '"': ss << "\\\""; break;
+            case '/': ss << "\\/"; break;
+            case '\b': ss << "\\b"; break;
+            case '\f': ss << "\\f"; break;
+            case '\n': ss << "\\n"; break;
+            case '\r': ss << "\\r"; break;
+            case '\t': ss << "\\t"; break;
+            default: ss << *iter; break;
+      }
+  }
+  ss << '"';
+  return ss.str();
+}
```

Comment:
Yeah, a fixed sized stack-variable, without any checks, we cannot use.

But, we can use a stack variable of size 1024 (for example) and before allocating we check to see whether the in-variable is shorter than 512. If so (probably 99% of the cases), we use the stack-variable and we avoid the malloc+free.

And that is the reason why a function like this should never make the decision to do an alloc, its users should always decide that.

If you look at common functions like 'strcpy' and alike, they don't allocate.
It's more, they don't even check whether the buffer to copy to is valid.
strcpy("123", NULL) gives a SIGSEGV.
Why? 
Well, it is not strcpy:s decision to do that check. We're talking about 3-4 assembler instructions here. That is 3-4 too many for some uses. Let the user decide, not the library. Important rules for library writing under C. I know these detail may seem almost silly for a Java or Python developer but in C it's different, it's the real thing. It is not uncommon to count assembler instructions or clock cycles while implementing ...